src/participant/participant.ts

```diff
@@ -387,39 +330,63 @@ export default class ParticipantController {
     return selectedQuickPickItem?.data;
   }
 
-  async selectCollectionWithParticipant(name: string): Promise<boolean> {
-    if (!name) {
-      const selectedName = await this._selectCollectionWithCommandPalette();
-      this._setCollectionName(selectedName);
-    } else {
-      this._setCollectionName(name);
+  async selectCollectionWithParticipant({
+    chatId,
+    databaseName,
+    collectionName: _collectionName,
+  }: {
+    chatId: string;
+    databaseName: string;
+    collectionName?: string;
+  }): Promise<boolean> {
+    let collectionName: string | undefined = _collectionName;
+    if (!collectionName) {
```

Comment:
I made them two variables since when the collection name isn't present we set it with the quick pick.
```
collectionName = await this._selectCollectionWithQuickPick(databaseName);
```
We could avoid the extra declaration, and set to the variable that's passed in. I didn't do that as I find setting to variables that are passed in arguments to also look a bit awkward. It wouldn't happen here, but I think it starts to bring in a pattern of possibly mutating the argument which we don't want. It would only be an issue with non-primitives, so again, not here.

Would you prefer if it's written as:
```ts
  async selectCollectionWithParticipant({
    chatId,
    databaseName,
    collectionName,
  }: {
    chatId: string;
    databaseName: string;
    collectionName?: string;
  }): Promise<boolean> {
    if (!collectionName) {
      collectionName = await this._selectCollectionWithQuickPick(databaseName);
      if (!collectionName) {
        return false;
      }
    }

```
I'm cool with either.